### PR TITLE
(WIP) fix: customize torch.guard_filter to fix missing logprobs recompilation

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

Currently, `torch.compiler.keep_tensor_guards_unsafe` is not working as expected.
Since the warmup runs without logprobs, requests with logprobs end up following a code path that does not generate logprobs.
https://github.com/rebellions-sw/vllm-rbln/actions/runs/20653949572/job/59303397534?pr=248

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`bug-fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/v1/worker/utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    # 1) Start from PyTorch's built-in tensor-guard keep policy
+    base_keep = torch.compiler.keep_tensor_guards_unsafe(guard_entries)
+
+    # 2) Add our extra keep condition
+    out = []
+    for keep, e in zip(base_keep, guard_entries):
+        name = getattr(e, "name", "") or ""
+        keep_max_num_logprobs = ("num_logprobs" in name)
+        out.append(bool(keep or keep_max_num_logprobs))
+    return out
```
[the remark this answers is not in the document]
Comment (rebel-eunji) [reply vllm_rbln/v1/worker/utils.py]: ref) https://github.com/pytorch/pytorch/blob/3b10638d4a07089ef5ce3b3082ff820417b7479f/torch/_dynamo/guards.py#L2343
